sdk/preview_termination_queries.go

```diff
@@ -0,0 +1,794 @@
+package sdk
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math"
+	"strings"
+	"time"
+
+	corebig "math/big"
+
+	"github.com/filecoin-project/go-state-types/big"
+	"github.com/filecoin-project/go-state-types/builtin/v9/miner"
+	"golang.org/x/exp/constraints"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-bitfield"
+	"github.com/filecoin-project/go-state-types/abi"
+	lotusapi "github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/types"
+	miner8 "github.com/filecoin-project/specs-actors/v8/actors/builtin/miner"
+	poolstypes "github.com/glifio/go-pools/types"
+	"github.com/glifio/go-pools/util"
+	"github.com/ipfs/go-cid"
+	cbor "github.com/ipfs/go-ipld-cbor"
+)
+
+// PreviewTerminateSector gets the burnt funds for when a single sector is terminated
+func (q *fevmQueries) PreviewTerminateSector(
+	ctx context.Context,
+	minerAddr address.Address,
+	tipset string,
```

Comment:
Wishlist - it would be nice if this was a proper tipset type so we can get assistance from the compiler when we mess up types